tests/common/dualtor/dual_tor_io.py

```diff
@@ -374,16 +374,17 @@ def traffic_sender_thread(self):
             time.sleep(self.send_interval)
             # the stop_early flag can be set to True by data_plane_utils to stop prematurely
             if self.stop_early:
-                time.sleep(5)
-                self.stop_sniffer_early()
-                logger.info("Stop the sender thread gracefully after sending {} packets"\
-                    .format(sent_packets_count))
                 break
             testutils.send_packet(self.ptfadapter, *entry)
             self.packets_sent_per_server[server_addr] =\
                 self.packets_sent_per_server.get(server_addr, 0) + 1
             sent_packets_count = sent_packets_count + 1
 
+        time.sleep(10)
+        self.stop_sniffer_early()
```

Comment:
Why is `stop_sniffer_early` moved out of `stop_early` check?

In the NON stop-early path, the join call is waiting for sniffer to either complete or timeout:
https://github.com/Azure/sonic-mgmt/blob/7c1626bd70e88490ef5617647a71f459921589d8/tests/common/dualtor/dual_tor_io.py#L446

Is this done to save test execution time? As soon as ALL packets are sent, this change will wait 10s and stop sniffer. What will happen if sniffer completes ideal execution (join completes) within 10s sleep here? I think we will end sending `pkill -SIGINT` to a process which doesn't exist.
It still shouldn't be an issue as `module_ignore_errors=True` is set in `stop_sniffer_early`.  But, I just want to make sure.

Reply:
Yes, this is done to decrease execution time. I increased the sniffer timeout to 5 minutes total, and in my experiments found that the traffic was always able to complete sending within 3 minutes or so. If we do see an error from sending the `SIGINT` then that would indicate that there is some problem delaying the traffic sender thread.